app/controllers/admin/dfc_product_imports_controller.rb

```diff
@@ -35,10 +35,18 @@ def index
       end
 
       @count = imported.compact.count
+    rescue Rack::OAuth2::Client::Error => e
+      flash[:error] = I18n.t(
+        'admin.dfc_product_imports.index.oauth_error_html',
+        message: e.message,
+        oidc_settings_link: ActionController::Base.helpers.link_to(
+          I18n.t('spree.admin.tab.oidc_settings'), Rails.application.routes.url_helpers.admin_oidc_settings_path
+        )
+      ).html_safe
```

Comment:
Hm, now a Keycloak server could inject HTML into our page. That server should be trusted, but still, maybe we can escape that one.

Reply:
😅  Good pickup, yes that should be escaped (or excluded).